### PR TITLE
Make 0-RTT explicit to the backend

### PR DESCRIPTION
**Pull request by joaosa**
tl;dr: TlS 1.3 0-RTT leaves applications vulnerable to replay attacks. There are also issues with forward secrecy, but there is a little we can do there.

One way to counter this is to explicitly signal one's backend that early data is being sent. This way, that backend can decide to drop a given request for non-idempotent operations.

I don't think this is an actual risk since, currently, we're only distributing static content.
This won't matter for a typical `curl https://strn.pl/ipfs/QmQ2r6iMNpky5f1m4cnm3Yqw8VSvjuKpTcK1X7dBR1LkJF/cat.gif`.

It's something important to keep in mind however. More info [here](https://blog.trailofbits.com/2019/03/25/what-application-developers-need-to-know-about-tls-early-data-0rtt/).

### CODE REVIEW

**Review by DiegoRBaquero — APPROVED** [review]
We don't really need this until we actually have such endpoints in the shim. Not a security risk right now, but on the side of prevention, what's the perf impact of just adding a dynamic header to proxy back? I am sure it's minimal, but is it negligible?

**Comment by joaosa** [comment]
Since we can't be entirely sure whether there's any measurable performance impact, let's close this for now and apply this when there's a usecase 👍
